sdk/core/Azure.Core/src/Shared/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using Azure.Core.Serialization;
+
+namespace Azure.Core
+{
+    internal static class Utf8JsonReaderExtensions
```

Comment:
Why use internal shared source rather than putting these on a public type?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/37970